src/Workspaces/Remote/ServiceHub/Services/SourceGeneration/RemoteSourceGenerationService.cs

```diff
@@ -132,4 +132,21 @@ protected override IRemoteSourceGenerationService CreateService(in ServiceConstr
 
         return false;
     }
+
+    public ValueTask<ImmutableArray<SourceGeneratorIdentity>> GetSourceGeneratorIdentitiesAsync(
+        Checksum solutionChecksum,
+        ProjectId projectId,
+        string analyzerReferenceFullPath,
+        CancellationToken cancellationToken)
+    {
+        return RunServiceAsync(solutionChecksum, solution =>
```

Comment:
remote side of this call is trivial.  we just retrieve hte AnalyzerFileRef requested, fetch its generators and get the identities from that.